service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/ip/DeleteAzureIpStep.java

```diff
@@ -68,7 +57,9 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
   @Override
   public StepResult undoStep(FlightContext flightContext) throws InterruptedException {
```

Comment:
I keep meaning to make a `OneWayStep` class that handles the erroring on undo, and/or a variant of that with a no-op undo, since we have to write this function unnecessarily in a lot of places.

Reply:
We should put that into Stairway. There are two undo cases that we code all the time:
- undo that returns success
- undo that triggers dismal failure
I'd like both of those as Step derivatives in Stairway